### PR TITLE
chore: open a PR instead of pushing for Snyk updates

### PR DESCRIPTION
**Pull request by crenshaw-dev**
This PR solves a few problems:

1. It allows us to get the Snyk docs updates into the `master` branch without using an App or some other hack to bypass the approver requirement.
2. It gives us a weekly reminder (in the form of a PR) to check the latest Snyk scan.
3. It adds the option to trigger one-off runs, which will be helpful for testing and for scanning immediately before major releases.
4. It adds support for RC releases.
5. It cleans up the [filenames with colons in them](https://github.com/argoproj/argo-cd/issues/10437) which cause problems for Windows users.

### CODE REVIEW

File: .github/workflows/update-snyk.yaml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 name: Snyk report update
 on:
+  workflow_dispatch: {}
```
Comment (crenshaw-dev) [inline .github/workflows/update-snyk.yaml:3]: Allow this to be run on-demand.

**Review by 34fathombelow — APPROVED** [review]
LGTM!
I love the idea of creating a PR rather than writing directly to master.

**Comment by crenshaw-dev** [comment]
@34fathombelow I wanna drop the `contents: write` permission entirely, and I think I'm gonna do that by creating a bot account. But that'll wait for another PR.  :-)

**Comment by 34fathombelow** [comment]
@crenshaw-dev could we use ```argo-bot``` like we do in the release workflow?

**Comment by crenshaw-dev** [comment]
Hm. I'm not sure that `argo-bot` is an account, I thought it was just a name... Looking

**Comment by crenshaw-dev** [comment]
Ah-ha! I bet I can use this: https://github.com/argo-bot

**Review by leoluz — APPROVED** [review]
LGTM

File: .github/workflows/update-snyk.yaml [hunk 2]
```diff
@@ -1,5 +1,6 @@
...
   schedule:
     - cron: '0 0 * * 0' # midnight every Sunday
```
Comment (34fathombelow) [inline .github/workflows/update-snyk.yaml:5]: FYI, GitHub cron times are UTC.  This cronjob would be equivalent to Saturday 5pm PST.  This actually works out great for me.

Comment (crenshaw-dev) [reply .github/workflows/update-snyk.yaml:5]: Haha yeah actually works pretty well for me too

**Pull request merged**